src/test/scala/edu/ie3/simona/agent/grid/DBFSAlgorithmParticipantSpec.scala

```diff
@@ -0,0 +1,353 @@
+                gridAgentWithParticipants
+              )
+            )
+          )
+        )
+      )
+
+    }
+
+    s"check the request asset power message indirectly" in {
+
+      val startGridSimulationTriggerId = 4
+      val firstSweepNo = 0
+
+      // send the start grid simulation trigger
+      // the gird agent should send a RequestAssetPowerMessage to the load agent
+      scheduler.send(
+        gridAgentWithParticipants,
+        TriggerWithIdMessage(
+          StartGridSimulationTrigger(3600L),
+          startGridSimulationTriggerId,
+          gridAgentWithParticipants
+        )
+      )
+
+      // we expect a request for voltage values of our slack node
+      // (voltages are requested by our agent under test from the superior grid)
+      val firstSlackVoltageRequestSender =
+        superiorGridAgent.expectSlackVoltageRequest(firstSweepNo)
+
+      // we now answer the request of our gridAgentsWithParticipants
+      // with a fake slack voltage message
+      superiorGridAgent.gaProbe.send(
+        firstSlackVoltageRequestSender,
+        ProvideSlackVoltageMessage(
+          firstSweepNo,
+          Seq(
+            ExchangeVoltage(
+              supNodeA.getUuid,
+              Quantities.getQuantity(380, KILOVOLT),
+              Quantities.getQuantity(0, KILOVOLT)
+            )
+          )
+        )
+      )
+
+      // power flow calculation should run now. After it's done,
+      // our test agent should now be ready to provide the grid power values,
+      // hence we ask for them and expect a corresponding response
+      superiorGridAgent.requestGridPower(
+        gridAgentWithParticipants,
+        firstSweepNo
+      )
+
+      // the gridAgentWithParticipants has received an AssetPowerChangedMessage
+      // before requesting power from the superiorGrid
+      superiorGridAgent.expectGridPowerProvision(
+        Seq(
+          ExchangePower(
+            supNodeA.getUuid,
+            Quantities.getQuantity(135.90837346741768, MEGAWATT),
+            Quantities.getQuantity(60.98643348675892, MEGAVAR)
+          )
+        )
+      )
+
+      // before the second sweep the gridAgentWithParticipants will receive an AssetPowerUnchangedMessage
+      // we start a second sweep by asking for next sweep values which should trigger the whole procedure again
+      val secondSweepNo = firstSweepNo + 1
+
+      superiorGridAgent.requestGridPower(
+        gridAgentWithParticipants,
+        secondSweepNo
+      )
+
+      // the agent now should ask for updated slack voltages from the superior grid
+      val secondSlackAskSender =
+        superiorGridAgent.expectSlackVoltageRequest(secondSweepNo)
+
+      // the superior grid would answer with updated slack voltage values
+      superiorGridAgent.gaProbe.send(
+        secondSlackAskSender,
+        ProvideSlackVoltageMessage(
+          secondSweepNo,
+          Seq(
+            ExchangeVoltage( // this one should currently be ignored anyways
```

Comment:
Why should this be ignored? Shouldn't the voltage of the superior grid connection node be considered with powerflow calculation?